src/main/java/dinkplugin/notifiers/DiaryNotifier.java

```diff
@@ -107,6 +107,11 @@ public void onMessageBox(String message) {
                 return;
             }
 
+            if (!checkDifficulty(difficulty)) {
```

Comment:
Could you move this check just before the `invokeLater` call so `diaryCompletionById` is updated even if a notification is not fired?

Reply:
All done, what's the point of this map? Wouldn't it be discarded on a new invocation of this notifier?

Reply:
thanks! `diaryCompletionById` is useful to avoid duplicate notifications (not relevant in this case), and to report the total number of diaries that have been completed (so a low-tier completion is still relevant for later high-tier completions): https://github.com/pajlads/DinkPlugin/blob/ad4d733d0cb45345fbdcf132b68b99f68fe292b8/src/main/java/dinkplugin/notifiers/DiaryNotifier.java#L189